src/main/scala/mesosphere/util/state/MesosLeaderInfo.scala

```diff
@@ -0,0 +1,30 @@
+package mesosphere.util.state
+
+/**
+  * Utility class for keeping track of a Mesos Leader
+  */
+trait MesosLeaderInfo {
+
+  def currentLeaderUrl: Option[String]
+
+  def load(host: String, port: Int): Unit
```

Comment:
Or better: `onNewMasterInfo` and pass the `MasterInfo` as argument. The then the logic of constructing an URL from that is inside this object.